legacy/nosql/mongo/mongo.go

```diff
@@ -527,15 +630,17 @@ func (u *Update) Upsert(d nosql.Document) nosql.Update {
 	return u
 }
 func (u *Update) Do(ctx context.Context) error {
-	key := compKey(u.key)
+	idFilter := bson.M{idField: compKey(u.key)}
 	var err error
+	updateOptions := options.Update()
 	if u.upsert != nil {
 		if len(u.upsert) != 0 {
+			updateOptions.SetUpsert(true)
 			u.update["$setOnInsert"] = u.upsert
 		}
-		_, err = u.col.c.UpsertId(key, u.update)
+		_, err = u.col.c.UpdateOne(ctx, idFilter, u.update, updateOptions)
 	} else {
-		err = u.col.c.UpdateId(key, u.update)
+		_, err = u.col.c.UpdateOne(ctx, idFilter, u.update, updateOptions)
```

Comment:
It's the same as the `UpdateOne` above, right? So we can remove the duplication an move it out of the `if` statement.

Reply:
yup, updated.